src/Compilers/CSharp/Portable/Symbols/TypeWithState.cs

```diff
@@ -28,6 +29,38 @@ public static TypeWithState Create(TypeSymbol type, NullableFlowState defaultSta
             return new TypeWithState(type, state);
         }
 
+        public static TypeWithState Create(TypeWithAnnotations typeWithAnnotations, FlowAnalysisAnnotations annotations = FlowAnalysisAnnotations.None)
+        {
+            var type = typeWithAnnotations.Type;
+            if (type is null)
+            {
```

Comment:
I don't understand the purpose of this `if` statement.  If it isn't necessary, please remove it.